src/main/java/net/sf/jabref/gui/EntryTypeDialog.java

```diff
@@ -148,6 +172,95 @@ private JPanel createEntryGroupPanel(String groupTitle, Collection<EntryType> en
         return panel;
     }
 
+    private JPanel createIdFetcherPanel() {
+        JButton generateButton = new JButton(Localization.lang("Generate"));
+        JButton cancelButton = new JButton(Localization.lang("Cancel"));
+        JTextField idTextField = new JTextField("");
+        JComboBox<String> comboBox = new JComboBox<>();
+        EntryFetchers.getIdFetchers().forEach(fetcher -> comboBox.addItem(fetcher.getName()));
+        JLabel fetcherLabel = new JLabel("ID type"), idLabel = new JLabel("ID");
+
+        SwingWorker<Optional<BibEntry>, Void> fetcherWorker = new SwingWorker<Optional<BibEntry>, Void>() {
+            Optional<BibEntry> bibEntry = Optional.empty();
+            IdBasedFetcher fetcher = null;
+            String searchID = "";
+
+            @Override
+            protected Optional<BibEntry> doInBackground() throws Exception {
+                searchID = idTextField.getText().trim();
+                fetcher = EntryFetchers.getIdFetchers().get(comboBox.getSelectedIndex());
+                if (!searchID.isEmpty()) {
+                    try {
+                        bibEntry = fetcher.performSearchById(searchID);
+                    } catch (FetcherException e) {
+                        LOGGER.error("Error fetching from " + fetcher.getName(), e);
+                        JOptionPane.showMessageDialog(null, Localization.lang("Error while fetching from %0", fetcher.getName()), Localization.lang("Error"), JOptionPane.ERROR_MESSAGE);
+                    }
+                }
+                dispose();
+                return bibEntry;
+            }
+
+            @Override
+            protected void done() {
+                if (isCancelled()) {
+                    JOptionPane.showMessageDialog(null, Localization.lang("%0 import canceled", fetcher.getName()),Localization.lang("Import canceled by user"), JOptionPane.INFORMATION_MESSAGE);
+                    return;
+                }
+
+                try {
+                    Optional<BibEntry> result = get();
+                    if (result.isPresent()) {
+                        frame.getCurrentBasePanel().insertEntry(result.get());
+                    } else {
+                        JOptionPane.showMessageDialog(null, Localization.lang("No entry with id '%0' for fetcher '%1' was found.", searchID, fetcher.getName()), Localization.lang("No files found."), JOptionPane.WARNING_MESSAGE);
+                    }
+                } catch (ExecutionException | InterruptedException e) {
+                    LOGGER.error("No entry with id '" + searchID + " for fetcher '" + fetcher.getName() + "' was found.");
```

Comment:
We changed that when we set together. Why did you change that back? Is it because of some rebase stuff? In general, to not rebase, but do merge.